integrationTests/testConsensusNode.go

```diff
@@ -0,0 +1,462 @@
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	dataBlock "github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/data/endProcess"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/hashing/blake2b"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	ed25519SingleSig "github.com/ElrondNetwork/elrond-go-crypto/signing/ed25519/singlesig"
+	mclsinglesig "github.com/ElrondNetwork/elrond-go-crypto/signing/mcl/singlesig"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/consensus/round"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/epochStart/metachain"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/factory/peerSignatureHandler"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/mock"
+	"github.com/ElrondNetwork/elrond-go/node"
+	"github.com/ElrondNetwork/elrond-go/ntp"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	syncFork "github.com/ElrondNetwork/elrond-go/process/sync"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	elrondShardingMocks "github.com/ElrondNetwork/elrond-go/sharding/mock"
+	"github.com/ElrondNetwork/elrond-go/sharding/nodesCoordinator"
+	"github.com/ElrondNetwork/elrond-go/state"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/lrucache"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+	"github.com/ElrondNetwork/elrond-go/storage/timecache"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/testscommon/cryptoMocks"
+	dataRetrieverMock "github.com/ElrondNetwork/elrond-go/testscommon/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/testscommon/nodeTypeProviderMock"
+	"github.com/ElrondNetwork/elrond-go/testscommon/shardingMocks"
+	statusHandlerMock "github.com/ElrondNetwork/elrond-go/testscommon/statusHandler"
+	vic "github.com/ElrondNetwork/elrond-go/testscommon/validatorInfoCacher"
+	"github.com/ElrondNetwork/elrond-go/trie"
+	"github.com/ElrondNetwork/elrond-go/trie/hashesHolder"
+)
+
+const (
+	blsConsensusType = "bls"
+	signatureSize    = 48
+	publicKeySize    = 96
+)
+
+var testPubkeyConverter, _ = pubkeyConverter.NewHexPubkeyConverter(32)
+
+// TestConsensusNode represents a structure used in integration tests used for consensus tests
+type TestConsensusNode struct {
+	Node             *node.Node
+	Messenger        p2p.Messenger
+	NodesCoordinator nodesCoordinator.NodesCoordinator
+	ShardCoordinator sharding.Coordinator
+	ChainHandler     data.ChainHandler
+	BlockProcessor   *mock.BlockProcessorMock
+	ResolverFinder   dataRetriever.ResolversFinder
+	AccountsDB       *state.AccountsDB
+	NodeKeys         TestKeyPair
+}
+
+// NewTestConsensusNode returns a new TestConsensusNode
+func NewTestConsensusNode(
```

Comment:
for simplicity, we might have removed maxShards and nodeShardId as the consensus is always tested intra shard. (could have been hardcoded through some consts to 1 and 0)